addons/popochiu/engine/objects/character/popochiu_character.gd

```diff
@@ -497,7 +497,13 @@ func queue_walk_to_clicked(offset := Vector2.ZERO) -> Callable:
 ## Makes the character walk to the last clicked [PopochiuClickable], which is stored in
 ## [member Popochiu.clicked]. You can set an [param offset] relative to the target position.
 func walk_to_clicked(offset := Vector2.ZERO) -> void:
+	var clicked_id: String = E.clicked.script_name
+	
 	await _walk_to_node(E.clicked, offset)
+	
+	# A double check in case two PopochiuClickable share the same `walk_to_point` coordinates
+	if clicked_id != E.clicked.script_name:
```

Comment:
Sorry man, but I don't get what are you doing here. I see nothing wrong, so I'm just curious. Can you please explain?

Reply:
Imagine a scenario where two props/hotspots/characters have equal values in their `walk_to_point` properties. Since the cancellation of movement depends on the character's final position matching the expected target position (`popochiu_character.gd:825`), the code execution would continue because the condition `if not is_blocking and position != target_move_pos` evaluates to `false`. To prevent this from happening, I added an additional check: that the object that initiated the character's movement is the same active object (`E.clicked`) once the movement has finished. I know that by doing this, we could simplify everything to just this single check (ignoring the position comparison). I just kept both because I wasn't sure... or because I'm an idiot.

If you think it's clearer to just leave this latter one, I could get rid of several things that raised doubts in other comments.

Reply:
Actually... I think I will leave only this one.

Reply:
Furthermore, there might be another option that would avoid having to compare the values of `script_name`: having a boolean property in the **PopochiuClickable** class (e.g., `execution_cancelled`) that is set to `true` when an interaction interrupts the execution of instructions (e.g., clicking on the scene or another object).

Reply:
I saw your last commit, and I think this is way cleaner.

To me this is mergeable!